weave/Scripts/Main.cs

```diff
@@ -224,18 +248,27 @@ private void OnPlayerReachedGoal()
         if (++_roundCompletions != _lobby.Count)
             return;
 
-        HandleRoundComplete();
-    }
-
-    private void HandleRoundComplete()
-    {
         _roundCompletions = 0;
-
         _scoreDisplay.OnRoundComplete();
+        StartPreparationPhase();
+    }
 
-        DisablePlayerMovement();
+    private void StartPreparationPhase()
+    {
         ClearLinesAndSegments();
         ClearAndSpawnGoals();
+        SetPlayerMovement(false);
+        _uiUpdateTimer.Timeout += UpdateCountdown;
+        _playerDelayTimer.Start();
+        _scoreDisplay.Enabled = false;
+
+        Task.Run(async delegate
```

Comment:
Could you do this logic with a timer? :)

Reply:
![image](https://github.com/weave-game/weave/assets/42991003/bb270af5-1f72-4f61-bb6f-c95b8e8e1887)